### DISCUSSION TITLE
Getting "Server Error" while trying to update listing inventory

### DISCUSSION THREAD

**Discussion by SrdjanPaunovic** [etsy/open-api#966] [Q&A]
I tried to update the listing inventory using API endpoint: `https://openapi.etsy.com/v3/application/listings/{listing_id}/inventory`

I had a large number of products (over 600-700 products). In most cases I got 'Server Error'. Is there some limitation on how many products I can add to the listing?

I found this information on [this site](https://www.etsysimplicity.com/how-many-variations-can-you-have-on-an-etsy-listing-a-simple-to-understand-guide/): 
```
You can have a maximum of 2 variant types (for example size, colour, material), with a maximum of 70 options between them.
This means you could have 2 variation properties of size and colour, with a total of 35 options per variation property.
```

If I understand this correctly I can have 35 different colors per 1 size (if I have color and size props), which means I can have 35*35 products on that listing.

1. If I'm wrong, can someone better explain what the limits are?
2. Can you guys have a better validation message there, some 400 errors with the appropriate message?

**Comment by hyyyper** [comment]
Your understanding of the options limitation is correct.
(35, 1) = 35 options, (35, 35) = 70 options.

Also, the "too many options" error looks different, so it must be something else 🤔

Can you provide an example of one of the product objects?
```
{
    "sku": string,
    "property_values": [ ... ],
    "offerings": [ ... ],
}
```


I would assume this is server error is unintended and probably being caused by an out-of-memory error on the service handling the request.

I have run into a similar issue when requesting lots of listings with lots of variants:
https://github.com/etsy/open-api/discussions/892

**Reply by SrdjanPaunovic** [reply]
Does that mean that I can't have more than 70 products per listing (product = options = variants)?

I tested a scenario where I have more than 35 different colors for the same size (also tried with more than 70 and still worked). So I can't understand how to limit/validate those values.

Here is an example you asked:
```
{
    "product_id": 16003843786,
    "sku": "MD-6000",
    "is_deleted": false,
    "offerings": [
        {
            "offering_id": 15957294036,
            "quantity": 999,
            "is_enabled": true,
            "is_deleted": false,
            "price": {
                "amount": 1300,
                "divisor": 100,
                "currency_code": "USD"
            }
        }
    ],
    "property_values": [
        {
            "property_id": 513,
            "property_name": "Size",
            "scale_id": null,
            "scale_name": null,
            "value_ids": [
                1150681902869
            ],
            "values": [
                "T-Shirt - XS"
            ]
        },
        {
            "property_id": 200,
            "property_name": "Primary color",
            "scale_id": null,
            "scale_name": null,
            "value_ids": [
                1160119550367
            ],
            "values": [
                "Black"
            ]
        }
    ]
}
```

**Reply by hyyyper** [reply]
product = variant
option != product

70 unique objects within property_values:
Black, Red, Green, T-Shirt - XS, T-Shirt - S = 5 options

If you're not having any success, perhaps try sending the minimum necessary data.
https://developers.etsy.com/documentation/reference/#operation/updateListingInventory

You could omit (if always null) scale_id, scale_name, is_deleted, and simplify the price to a single float.

Not sure if this will help.

**Reply by SrdjanPaunovic** [reply]
That's maybe the problem because I am able to send more than 70 unique objects. It is working ok if I have a few options. I just want to know what are the real limits, so I can validate my input before I send it to Etsy.
